src/OpenTelemetry/ServiceProviderExtensions.cs

```diff
@@ -0,0 +1,28 @@
+using System;
+
+namespace OpenTelemetry
+{
+    public static class ServiceProviderExtensions
+    {
+        private static readonly Type OptionsGeneircType = Type.GetType("Microsoft.Extensions.Options.IOptions`1, Microsoft.Extensions.Options");
+
+        public static T GetOptions<T>(this IServiceProvider serviceProvider)
+            where T : class, new()
+        {
+            if (OptionsGeneircType == null)
+            {
+                throw new InvalidOperationException("Microsoft.Extensions.Options.IOptions<> could not be found reflectively in the running process.");
+            }
+
+            Type optionsType = OptionsGeneircType.MakeGenericType(typeof(T));
+
+            object options = serviceProvider.GetService(optionsType);
+            if (options == null)
+            {
+                return new T();
+            }
+
+            return (T)optionsType.GetProperty("Value").GetMethod.Invoke(options, null);
+        }
```

Comment:
Again, we wouldn't need this if the core SDK supported factory methods for configuration because you could just register your exporter on that platform with a factory method that used the native API to get configuration options when new'ing up the exporter or instrumentation instance.